src/thir.md

```diff
@@ -40,7 +48,176 @@ which is useful to keep peak memory in check. Having a THIR representation of
 all bodies of a crate in memory at the same time would be very heavy.
 
 You can get a debug representation of the THIR by passing the `-Zunpretty=thir-tree` flag
-to `rustc`.
+to `rustc`. Here is how a function with just the statement `let x = 1 + 2;` gets represented in
+annotated THIR:
```

Comment:
This wording is kind of confusing, you can understand it as "annotated THIR" being a separate concept. I don't think we need to mention that annotations have been added
```suggestion
THIR:
```